src/atem.ts

```diff
@@ -554,4 +600,85 @@ export class Atem extends EventEmitter {
 			delete this._sentQueue[trackingId]
 		}
 	}
+
+	private _calcActiveMeInputs (mode: 'program' | 'preview', meId: number): number[] {
+		const inputs = new Set<number>()
+		const meRef = this.state.video.getMe(meId)
+		inputs.add(meRef[mode === 'program' ? 'programInput' : 'previewInput'])
+
+		// Upstream Keyers
+		Object.values(meRef.upstreamKeyers).filter(usk => {
+			if (mode === 'program') {
+				return usk.onAir
+			}
+
+			// Pretty gross bitwise operations in this next line, be warned.
+			// TODO(Lange): This feels fragile but I'm unsure how to make it more robust.
+			return meRef.transitionProperties.selection & (2 ** (usk.upstreamKeyerId + 1))
+		}).forEach(usk => {
+			inputs.add(usk.fillSource)
+
+			// This is the only USK type that actually uses the cutSource.
+			if (usk.mixEffectKeyType === Enums.MixEffectKeyType.Luma) {
+				inputs.add(usk.cutSource)
+			}
+		})
+
+		// DSKs only show up on ME 1,
+		// so we only add them if that's the ME we are currently processing.
+		// TODO: Is this still true for the new Constellation ATEMs?
```

Comment:
Someone who knows more about the new ATEMs needs to fact-check this.

Reply:
Yes, I believe this is true. I can't see the output, but they all show up only on the ME1 control page.